pkg/controller/usersignup/usersignup_controller_test.go

```diff
@@ -64,16 +66,19 @@ func TestUserSignupWithAutoApproval(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, reconcile.Result{}, res)
 
+	// Lookup the user signup again
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Name, Namespace: req.Namespace}, userSignup)
+	require.NoError(t, err)
+
 	mur := &v1alpha1.MasterUserRecord{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Spec.CompliantUsername, Namespace: req.Namespace}, mur)
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Status.CompliantUsername, Namespace: req.Namespace}, mur)
```

Comment:
at this point, you shouldn't use `userSignup.Status.CompliantUsername` as it is still empty - it will be present after the second reconcile.
So you have a few options:
* list all MURs and verify that there is only one
* list MUR with the label selector
* use the expected name of the MUR: `"foo-at-redhat-com"`

Reply:
btw, it would be good to have the two reconcile calls in two separated tests - it would be easier to read, better to verify (as it is isolated) and check when anything is broken

Reply:
Good catch here.. i've modified this test (and the others you mentioned) to return a list of MURs instead.